nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -339,6 +339,11 @@ public static boolean isRecordTypeCompatible(final Object value) {
             return dest;
         }
 
```

Comment:
Why wouldn't this be any List implementation?

Reply:
I have updated the PR.  Thanks for the good idea!